azurelinuxagent/common/protocol/wire.py

```diff
@@ -220,6 +220,19 @@ def _build_health_report(incarnation, container_id, role_instance_id, # pylint:
                          status, substatus, description):
     # Escape '&', '<' and '>'
     description = saxutils.escape(ustr(description))
+
+    # The max description that can be sent to WireServer is 4096 bytes.
+    # Exceeding this max can result in a failure to report health.
+    max_description_length = 4096
+    current_description_length = len(description)
+    if current_description_length > max_description_length:
+        logger.info(
+            'Trimmed health report description by {0} characters'.format(
+                current_description_length - max_description_length
+            )
+        )
+        description = description[-max_description_length:]
```

Comment:
the truncation is being done after encoding the xml string... this has the potential of producing invalid xml if the encoding for a charater is truncated

Reply:
Good catch, you are correct. I've moved this trimming logic to before encoding to prevent this problem.